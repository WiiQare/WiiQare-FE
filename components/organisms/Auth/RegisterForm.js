import React from "react";
import { createContext, useState } from "react";

import { StepLabel, Stack } from "@mui/material";
import Stepper from "@mui/material/Stepper";
import Step from "@mui/material/Step";
import StepConnector, {stepConnectorClasses} from "@mui/material/StepConnector";
import { styled } from "@mui/material/styles";

import MenuHolder from "../../atoms/MenuHolder";
import BlinkSnackbar from "../../atoms/BlinkSnackbar";
import StepRegistration from "./StepRegistration";

export const FormContextRegister = createContext();

const steps = [
	"Votre adresse email",
	"VÃ©rification de l'email",
	"Information personnel",
];

const ColorlibConnector = styled(StepConnector)(({ theme }) => ({
	[`&.${stepConnectorClasses.alternativeLabel}`]: {
		top: 11,
		left: "calc(-50% + 12px)",
		right: "calc(50% + 12px)",
	},
	[`&.${stepConnectorClasses.active}`]: {
		[`& .${stepConnectorClasses.line}`]: {
			backgroundColor: theme.palette.primary.main,
		},
	},
	[`&.${stepConnectorClasses.completed}`]: {
		[`& .${stepConnectorClasses.line}`]: {
			backgroundColor: theme.palette.primary.main,
		},
	},
	[`& .${stepConnectorClasses.line}`]: {
		height: 4,
		border: 0,
		backgroundColor:
			theme.palette.mode === "dark" ? theme.palette.grey[800] : "#f0f4fd",
		borderRadius: 1,
	},
}));

const RegisterForm = () => {
	const [activeStep, setActiveStep] = React.useState(0);
	const [completed, setCompleted] = React.useState({});
    const [formData, setFormData] = useState({});

	const totalSteps = () => {
		return steps.length;
	};

	const completedSteps = () => {
		return Object.keys(completed).length;
	};

	const isLastStep = () => {
		return activeStep === totalSteps() - 1;
	};

	const allStepsCompleted = () => {
		return completedSteps() === totalSteps();
	};

	const handleNext = () => {
		const newActiveStep =
			isLastStep() && !allStepsCompleted()
				? // It's the last step, but not all steps have been completed,
				// find the first step that has been completed
				steps.findIndex((step, i) => !(i in completed))
				: activeStep + 1;
		setActiveStep(newActiveStep);
	};

	const handleBack = () => {
		setActiveStep((prevActiveStep) => prevActiveStep - 1);
	};

	const handleStep = (step) => () => {
		setActiveStep(step);
	};

	const handleComplete = (values) => {
		const newCompleted = completed;
		newCompleted[activeStep] = true;
		setCompleted(newCompleted);
		handleNext();
	};

	return (
		<>
			<div className="form-holder">
				<MenuHolder href="/login" label="S'IDENTIFIER" />
				<div className="signin-signup-form">
					<Stack
						sx={{ mb: 2, mt: 2, width: "100%" }}
						justifyContent="center"
						alignItems="center"
					>
						<Stepper
							sx={{ maxWidth: 450, width: "100%" }}
							alternativeLabel
							activeStep={activeStep}
							connector={<ColorlibConnector />}
						>
							{steps.map((label, index) => (
								<Step key={label} completed={completed[index]}>
									<StepLabel color="inherit">
										{label}
									</StepLabel>
								</Step>
							))}
						</Stepper>
					</Stack>

					<div className="form-items">

						<FormContextRegister.Provider value={{ activeStep, setActiveStep, handleComplete, formData, setFormData }}>
							<StepRegistration />
						</FormContextRegister.Provider>

					</div>
				</div>
			</div>
			{/* <BlinkSnackbar /> */}
		</>
	);
}

export default RegisterForm;
