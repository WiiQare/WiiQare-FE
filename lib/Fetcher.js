import useSWR from 'swr';

const baseURL = "https://api.wiiqare-app.com/api/v1";

const Fetcher = (endpoint, token = null) => {

    console.log(token);

    const response = (...args) => fetch(...args, { headers: { 'Authorization': token ? `Bearer ${token}` : null } }).then(res => res.json())

    const { data, error, isLoading } = useSWR(`${baseURL}${endpoint}`, response)

    return {
        data,
        isLoading,
        isError: error
    }
}

export default Fetcher;