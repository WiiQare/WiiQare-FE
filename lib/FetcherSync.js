import useSWR from "swr";

const baseURL = "https://xxxx.com";

const FetcherSync = () => {
  const postRequest = (endpoint, token = null) => {
    const response = (...args) =>
      fetch(...args, {
        headers: { Authorization: token ? `Bearer ${token}` : null },
      }).then((res) => res.json());

    const { data, error, isLoading } = useSWR(
      `${baseURL}${endpoint}`,
      response
    );

    return { data, error, isLoading };
  };

  return {
    postRequest,
  };
};

export default FetcherSync;
