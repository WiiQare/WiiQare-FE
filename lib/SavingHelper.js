/* comment code below in case not using stripe */

const stripe = require('stripe')(process.env.STRIPE_SECRET_KEY);

/* ^^^^ comment code above in case not using stripe ^^^^ */

const PaymentStub = {
    paymentIntents: {
        create: async ( options ) => {
            return {
                client_secret: 'client-secret',
            }
        },
    }
};

const paymentLib = typeof stripe !== 'undefined'? stripe : PaymentStub;

export const createPaymentIntent = async ( options ) => {
    return await paymentLib.paymentIntents.create( options );
}

export const GenericPaymentExport =  typeof stripe != 'undefined'? paymentLib : paymentLib;
