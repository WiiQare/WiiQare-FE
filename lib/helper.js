const BASE_URL = `${process.env.NEXT_PUBLIC_API_URL}/api/v1`;

export async function sendEmail(formData) {
  try {
    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(
      BASE_URL + '/session/email-verification',
      Options,
    );

    console.log(response);
    const json = response.status == 200 ? response : await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function sendOtp(formData) {
  try {
    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(
      BASE_URL + '/session/email-validate-otp',
      Options,
    );
    const json = await response.json();

    console.log('test', json);
    return json;
  } catch (error) {
    return error;
  }
}

export async function register(formData) {
  try {
    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(BASE_URL + '/payer', Options);
    const json = await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function login(credentials) {
  try {
    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        email: credentials.email,
        password: credentials.password,
      }),
    };

    const response = await fetch(BASE_URL + '/session', Options);
    const json = await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function autocomplete(phone, token) {
  try {
    const Options = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
    };

    const response = await fetch(`${BASE_URL}/payer/patient/${phone}`, Options);
    const json = await response.json();

    console.log(json);
    return json;
  } catch (error) {
    return error;
  }
}

export async function savePatient(formData) {
  try {
    console.log(formData);
    let token = formData.accessToken;

    delete formData.accessToken;

    const Options = {
      method: formData?.id? 'PUT' : 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(BASE_URL + '/payer/patient', Options);
    const json = await response.json();

    console.log('save', json);
    return json;
  } catch (error) {
    return error;
  }
}

export async function convertCurrency(from, amount, to) {
  try {
    const Options = {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
        apikey: `i6sKwItIpG9o1PLxB6nykJ5OFecFsW8X`,
      },
    };

    const response = await fetch(
      `https://api.apilayer.com/exchangerates_data/convert?to=${to}&from=${from}&amount=${amount}`,
      Options,
    );
    const json = await response.json();

    console.log(json);
    return json;
  } catch (error) {
    return error;
  }
}

export async function sendEmailReset(formData) {
  try {
    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(BASE_URL + '/session/reset-password', Options);

    const json = response.status == 200 ? response : await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function resetPassword(formData) {
  try {
    console.log(formData);
    const Options = {
      method: 'PUT',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(BASE_URL + '/session/reset-password', Options);
    const json = await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function sendMessage(formData) {
  try {
    let token = formData.accessToken;

    delete formData.accessToken;

    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(BASE_URL + '/messaging/send', Options);
    const json = await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function invitationFriend(formData) {
  try {
    let token = formData.accessToken;

    delete formData.accessToken;

    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(BASE_URL + '/payer/send-invite', Options);
    const json =
      response.status == 201 ? { state: true } : await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function sendSMSHash(formData) {
  try {
    let token = formData.accessToken;

    delete formData.accessToken;

    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(BASE_URL + '/payer/send-sms-voucher', Options);

    console.log(response);
    const json =
      response.status == 200 || response.status == 201
        ? response
        : await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function addPlan(formData) {
  try {
    let token = formData.accessToken;

    delete formData.accessToken;

    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(
      `${process.env.NEXT_PUBLIC_API_URL}/api/v1/savings`,
      Options,
    );
    const json = await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

export async function saveOperation(formData) {
  try {
    let token = formData.accessToken;

    delete formData.accessToken;

    const Options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        Authorization: `Bearer ${token}`,
      },
      body: JSON.stringify(formData),
    };

    const response = await fetch(
      `${process.env.NEXT_PUBLIC_API_URL}/api/v1/operations`,
      Options,
    );
    const json = await response.json();

    return json;
  } catch (error) {
    return error;
  }
}

