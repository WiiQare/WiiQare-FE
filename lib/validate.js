export const loginValidate = async (values) => {
    const errors = {};

    if (!values.email) {
        errors.email = "Adresse email est requis"
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
        errors.email = 'Adresse email invalide';
    }

    if (!values.password) {
        errors.password = "Mot de passe est requis"
    } else if(values.password.includes(" ")) {
        errors.password = 'Mot de passe invalide';
    }

    return errors
}

export const registerValidate = async (values) => {
    const errors = {};

    if (!values.email) {
        errors.email = "Adresse email est requis"
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
        errors.email = 'Adresse email invalide';
    }

    if (!values.password) {
        errors.password = "Mot de passe est requis"
    } else if (values.password.length <= 8) {
        errors.password = 'Mot de passe trop court';
    } else if(values.password.includes(" ")) {
        errors.password = 'Mot de passe invalide';
    }

    if (values.cpassword !== values.password) {
        errors.cpassword = "Mot de passe de confirmation différent"
    }

    if (!values.username) {
        errors.username = 'Nom d\'utilisateur est requis';
    } else if (values.username.length <= 3) {
        errors.username = 'Le nom d\'utilisateur doit avoir au délà de 3 caractères';
    }

    return errors
}

export const emailValidate = async (values) => {
    const errors = {};

    if (!values.email) {
        errors.email = "Adresse email est requis"
    } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) {
        errors.email = 'Adresse email invalide';
    }

    return errors
}