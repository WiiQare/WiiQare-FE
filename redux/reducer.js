import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  client: { register: {}, patient: {} },
};

export const ReducerSlice = createSlice({
  name: 'wiiqare',
  initialState,
  reducers: {
    setRegister: (state, action) => {
      state.client.register = { ...action.payload };
    },
    setPatientDispatch: (state, action) => {
      state.client.patient = { ...action.payload };
    },
  },
});

export const { setRegister, setPatientDispatch } = ReducerSlice.actions;

export default ReducerSlice.reducer;
