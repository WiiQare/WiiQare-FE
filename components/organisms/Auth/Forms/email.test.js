import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Email from "./email";
import { FormContextRegister } from "../RegisterForm";
import { Provider } from "react-redux";
import { store } from "../../../../redux/store";
import { QueryClientProvider, QueryClient } from "react-query";
import * as Helpers from "../../../../lib/helper";

describe("Email", () => {
  let container;
  beforeEach(() => {
    jest.clearAllMocks();
    const queryClient = new QueryClient();
    const res = render(
      <QueryClientProvider client={queryClient}>
        <Provider store={store}>
          <FormContextRegister.Provider
            value={{
              activeStep: 0,
              setActiveStep: jest.fn(),
              handleComplete: jest.fn(),
              formData: {},
              setFormData: jest.fn(),
            }}
          >
            <Email />
          </FormContextRegister.Provider>
        </Provider>
      </QueryClientProvider>
    );
    container = res.container;
  });
  it("renders", () => {
    expect(container).toMatchSnapshot();
  });

  it("renders the form", () => {
    expect(container.querySelector("form")).toBeInTheDocument();
  });

  it("renders the email input", () => {
    expect(screen.getByLabelText("E-mail Address")).toBeInTheDocument();
  });

  it("should send email", async () => {
    jest.spyOn(Helpers, "sendEmail").mockResolvedValueOnce({});
    const user = userEvent.setup();
    const email = "test@example.com";
    const emailInput = screen.getByLabelText("E-mail Address");
    const form = container.querySelector("form");
    const submitButton = screen.getByRole("button", {
      name: "L'ÉTAPE SUIVANTE",
    });
    expect(submitButton).toBeInTheDocument();
    expect(emailInput).toBeInTheDocument();
    expect(form).toBeInTheDocument();
    expect(emailInput).toHaveValue("");
    await user.type(emailInput, email);
    expect(emailInput).toHaveValue(email);
    await user.click(submitButton);
    expect(Helpers.sendEmail).toHaveBeenCalledTimes(1);
    expect(Helpers.sendEmail).toHaveBeenCalledWith({ email });
  });

  it("should handle error if email is not sent", async () => {
    jest.spyOn(Helpers, "sendEmail").mockResolvedValueOnce({
      code: 400,
      message: "Email failed to send",
    });
    const user = userEvent.setup();
    const email = "test@example.com";
    const emailInput = screen.getByLabelText("E-mail Address");
    const submitButton = screen.getByRole("button", {
      name: "L'ÉTAPE SUIVANTE",
    });

    await user.type(emailInput, email);
    await user.click(submitButton);
    expect(Helpers.sendEmail).toHaveBeenCalledTimes(1);
    expect(Helpers.sendEmail).toHaveBeenCalledWith({ email });
  });
});
