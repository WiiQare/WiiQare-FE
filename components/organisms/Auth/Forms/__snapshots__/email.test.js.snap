// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Email renders 1`] = `
<div>
  <div
    class="MuiBox-root css-1vdpy2o"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
    >
      Entrez votre email
    </p>
  </div>
  <form
    id="signupform"
  >
    <div
      class="css-1p5q5e5-MuiStack-root"
    >
      <div
        class="space-y-1"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root lowercase css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="outlined-basic"
            id="outlined-basic-label"
          >
            E-mail Address
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              name="email"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  E-mail Address
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="form-button"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-jh47zj-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          L'Ã‰TAPE SUIVANTE
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
